src/python/pants/core/goals/test.py

```diff
@@ -143,6 +145,7 @@ def from_fallible_process_result(
         xml_results: Snapshot | None = None,
         extra_output: Snapshot | None = None,
         log_extra_output: bool = False,
+        all_results: List[FallibleProcessResult] | None = None,
```

Comment:
IMO, you should go ahead and deprecate the old argument in favor of this one (while updating all built in consumers). That would send a signal to anyone who has implemented test runners off-`main`, and give them a hint of what to do to add retry.

Reply:
sounds good, done. So far I've just boxed the results. I think rolling out retries to all the other backends can be deferred.